run-cgprna/run_cgprna/map.py

```diff
@@ -0,0 +1,135 @@
+import os
+import sys
+import re
+import shutil
+import fnmatch
+from string import Template
+from . import run_templates_in_shell, untar, mkdir
+
+STAR_MAP_TEMPLATE = Template('star_mapping.pl -s $sample_name -o $out_dir -t $threads -r $reference_data_root -sp $species -rb $ref_build -gb $gene_build -g $gene_build_gtf_name $other_options $raw_reads_string')
+MARK_DUPS_TEMPLATE = Template('bammarkduplicates2 I=$out_dir/$sample_name.star.Aligned.out.bam O=$out_dir/$sample_name.bam md5=1 index=1 markthreads=$threads md5filename=$out_dir/$sample_name.bam.md5 indexfilename=$out_dir/$sample_name.bam.bai M=$out_dir/$sample_name.bam.met tmpfile=$out_dir/biormdup')
+BAM_INDEX_TEMPLATE = Template('bamindex < $out_dir/$sample_name.star.AlignedtoTranscriptome.out.bam > $out_dir/$sample_name.star.AlignedtoTranscriptome.out.bam.bai')
+
+
+def map_seq_files(args):
+    '''
+    Top level entry point for mapping RNA-Seq sequence files.
+    '''
+    # keys should be the same as what they have in command_line.py
+    ref_related_args = {
+        '--species': args.species,
+        '--reference-build': args.ref_build,
+        '--gene-build': args.gene_build,
+        '--gene-build-gtf-name': args.gene_build_gtf_name
+    }
+
+    # only because star_mapping.pl will try to find files in a particular structure
+    ref_related_defaults = {
+        '--species': 'unspecified_species',
+        '--reference-build': 'unspecified_ref_build',
+        '--gene-build': 'unspecified_gene_build'
+    }
+
+    # only use temp_dir when needed to extract reference files
+    temp_dir = os.path.join(os.path.abspath(args.out_dir), 'cgpRna_map_temp')
+    clean_temp = 0
+
+    other_options = []
+    if args.rg_id_tag:
+        other_options.append('-lane-id "%s"' % args.rg_id_tag)
+    if args.lb_tag:
+        other_options.append('-library "%s"' % args.lb_tag)
+    if args.ds_tag:
+        other_options.append('-ds-tag "%s"' % args.ds_tag)
+    if args.pl_tag:
+        other_options.append('-machine-type "%s"' % args.pl_tag)
+    if args.pu_tag:
+        other_options.append('-npg-run "%s"' % args.pu_tag)
+
+    # prepare the output dir
+    mkdir(args.out_dir)
+
+    # prepare the reference
+    reference_data_root = os.path.abspath(args.ref)
+    # Anything not a file will be treated as a reference root folder
+    if os.path.isfile(reference_data_root):
+        # If a pre-built ref bundle tar file is supplied
+        if re.match(r'.*\.tar\.gz$', os.path.basename(reference_data_root)):
+            mkdir(temp_dir)
+            clean_temp = 1
+
+            # use the tar file name as the ref_root_dir_name, so that in BAM header, people can tell which bundle file was used
+            ref_root_dir_name = re.match(r'(.*)\.tar\.gz$', os.path.basename(reference_data_root)).group(1)
```

Comment:
Would not this capture the file name as well? Maybe just os.dirname?

Reply:
hmm, the variable name is a bit confusing. The line is trying to capture tar file base name without the extensions, so that the variable can be later on used to name the folder where reference files are extracted into.